docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,698 @@ id: pillars-turbomodules
+                    └── CalculatorPackage.java
+```
+
+##### AndroidManifest.xml
+
+```xml title="AndroidManifest.xml"
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.RTNCalculator">
+</manifest>
+```
+
+This is a small manifest file that defines the package for our module.
+
+
+##### CalculatorModule.java
+
+```java title="CalculatorModule.java"
+package com.RTNCalculator;
+
+import androidx.annotation.NonNull;
+import com.facebook.react.bridge.NativeModule;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import java.util.Map;
+import java.util.HashMap;
+
+public class CalculatorModule extends NativeCalculatorSpec {
+
+    public static String NAME = "RTNCalculator";
+
+    CalculatorModule(ReactApplicationContext context) {
+        super(context);
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public void add(double a, double b, Promise promise) {
+        promise.resolve(a + b);
+    }
+}
+```
+
+This class implements the module itself, which extends the `NativeCalculatorSpec` that was generated from the `NativeCalculator` JavaScript specification file.
+
+##### CalculatorPackage.java
+
+```java title="CalculatorPackage.java"
+package com.RTNCalculator;
+
+import androidx.annotation.Nullable;
+import com.facebook.react.bridge.NativeModule;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.module.model.ReactModuleInfo;
+import com.facebook.react.module.model.ReactModuleInfoProvider;
+import com.facebook.react.TurboReactPackage;
+import com.facebook.react.uimanager.ViewManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CalculatorPackage extends TurboReactPackage {
+
+  @Nullable
+  @Override
+  public NativeModule getModule(String name, ReactApplicationContext reactContext) {
+      if (name.equals(CalculatorModule.NAME)) {
+          return new CalculatorModule(reactContext);
+      } else {
+          return null;
+      }
+  }
+
+
+  @Override
+  public ReactModuleInfoProvider getReactModuleInfoProvider() {
+      return () -> {
+          final Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
+          moduleInfos.put(
+                  CalculatorModule.NAME,
+                  new ReactModuleInfo(
+                          CalculatorModule.NAME,
+                          CalculatorModule.NAME,
+                          false, // canOverrideExistingModule
+                          false, // needsEagerInit
+                          true, // hasConstants
+                          false, // isCxxModule
+                          false // isTurboModule
```

Comment:
I think this should be `true`. @cortinico what do you think?

Reply:
Yup correct, this should be fixed